### DISCUSSION TITLE
iBiblia v1.0.0.66

### DISCUSSION THREAD

**Discussion by rubiot** [rubiot/ibiblia#91] [General]
# iBiblia v1.0.0.66

This release includes the following changes:

### True interlinear mode
Added support for a true interlinear mode in chapter view. The user can choose between the old INTRAlinear mode (the one supported by theWord) and the new INTERlinear mode from the context menu.
![image](https://user-images.githubusercontent.com/8127450/145419682-35d3aac4-e5e5-4773-8540-9c48ea9a3c92.png)

### Better verse rules support for interlinear tab
The user can create independent verse rules for each interlinear layout and also have the option to disable rules and to choose if a rule is to be applied at view time only or at the export time.
Added code to detect verse rule errors, giving the user a chance to fix them.
![image](https://user-images.githubusercontent.com/8127450/145419928-168e6617-de8c-4b8d-ae67-44b0cb014336.png)

### Support for double click in chapter view, just like the verse edits, displaying the raw text in the export format.
![image](https://user-images.githubusercontent.com/8127450/145420421-7ed7105d-8597-4ce4-9dc4-b14da01bb034.png)
This is very useful in creating verse rules for the interlinear view, because it is possible to see their effect. Also, it is possible to copy-and-past the raw text to an external editor with good regex support.

### Support for partial modules
Now it is possible to import partial modules, for instance, import an .nt or a .ot module into a whole-Bible project.
### Added warning about importing non-UTF8 modules
iBiblia does not support non-UTF8 modules, so there's a good chance they would result in crashes and encoding errors.
### New verse rules dialog
Using a grid instead of the old checkboxes.
![image](https://user-images.githubusercontent.com/8127450/145421070-e1e37eda-7601-46f7-8934-9c3752e010c1.png)

### Added option to vaccuum project
There are some operations that can increase the size of a project disproportionally, such as applying patches and loading new texts. Now there is a new option to vaccuum the project.
![image](https://user-images.githubusercontent.com/8127450/145420549-36ac538f-398f-4805-b206-7c3277b5b749.png)

### Improvement in loading/creation times of big patches.
Patch creation/applying windows should load much faster now when dealing with big patches.

### Patch apply now ignores unchanged verses
When a patch is loaded iBiblia now first compares its content with the current project and discards unmodified verses, thus making clear what changes will be actually applied.

### Added visual indication of what parts of the verse are different on the patch.
Added labels to the bottom left of the windows indicating differences in red.
![image](https://user-images.githubusercontent.com/8127450/145421679-96f6b6e6-ce66-42e8-b34d-f70548f3dfc3.png)

### Initial support for right-to-left languages
Added right-to-left option to the verse context menu. The option is not persisted yet, so the user needs to set it every time a project is open. This is because the support is still rudimentary, with no suppport for mixed ltr/rtl text.
![image](https://user-images.githubusercontent.com/8127450/145422519-55c086c2-fcfd-42d4-b2e8-327b77678e40.png)

### Automatic system language detection on first execution
The language still can be configured in the *Options->Language* menu.

### iBiblia now detects new instances of theWord using a different language
The synchronization now works OK if the user opens a new theWord instance with a different language without having to restart iBiblia. The user still would have to restart theWord, because it does not persist the change to the config file right away, but only does it when closed.

### Ignoring associated punctuation in concordances and non-interlinear modules
This only makes sense for interlinears.

### Monospaced font on verse edit
Using a fixed-size font for the verse edit, instead of using the current verse font, which made little sense most of the time.
![image](https://user-images.githubusercontent.com/8127450/145422717-13fadb03-5499-4dcd-b3f3-f3cf7f6f5685.png)

### Support for <X>, <G> and <H> tags in chapter view
These are custom MySword interlinear tags.

### Fixed output for special cases of associated italic words
When the last associated word of a group was an italic word, the closing <Fi> tag ended up after the Strong's/morphology tags.

### Immediate theWord -> iBiblia syncronization
When the option "Sync theWord to iBiblia's verse" was enabled, the synchronization would happen only after the next verse change. Now it happens immediately, as one would expect.

### Verse rules with comments were not recognized in module importation.

<hr /><em>This discussion was created from the release <a href='https://github.com/rubiot/ibiblia/releases/tag/'>iBiblia v1.0.0.66</a>.</em>

**Comment by SalitaNgDiyos** [comment]
Always nice to see some new updates. Happy New Year, bro! 

Edmund
